ocp_utilities/infra.py

```diff
@@ -81,6 +85,46 @@ def assert_pods_failed_or_pending(pods):
         )
 
 
+def assert_nodes_in_healthy_condition(nodes):
+    """
+    Validates nodes are in a healthy condition.
+    Nodes Ready condition is True and none of the following node conditions is True:
+        - DiskPressure
+        - MemoryPressure
+        - PIDPressure
+        - NetworkUnavailable
+
+    Args:
+         nodes(list): List of Node objects
+
+    Raises:
+        NodesNotHealthyConditionError: if any nodes DiskPressure
+        MemoryPressure, PIDPressure, NetworkUnavailable condition is True
+    """
+    LOGGER.info("Verify all nodes are in a healthy condition")
+    unhealthy_nodes_with_conditions = {}
+    for node in nodes:
+        condition_type_list = []
+        for condition in node.instance.status.conditions:
+            if (
+                condition.type != Node.Condition.READY
+                and condition.status == Node.Condition.Status.TRUE
+            ):
+                condition_type_list.append(condition.type)
+
+        if condition_type_list:
+            unhealthy_nodes_with_conditions[node.name] = condition_type_list
+
```

Comment:
```suggestion
            if condition_type_list:
                unhealthy_nodes_with_conditions[node.name] = condition_type_list

```

Reply:
@rnetser, is there a compelling reason why we would need to indent the `if `code segment `if condition_type_list:`? If this `if` statement is indented, then it will be called for each node condition - 5 times for every node. The way how it is now, it is only called once - after the `for condition in node.instance.status.conditions` loop.

Reply:
meant to add list comprehension, so for each node, have 1 line which gathers the failed conditions and then add to dict